### PR TITLE
feat(compiler): integrate compiler with view engine - main integratio… …n tests work

### PR DESCRIPTION
**Pull request by tbosch**
First commit is from https://github.com/angular/angular/pull/14237

### CODE REVIEW

File: modules/@angular/compiler/src/metadata_resolver.ts [hunk 1]
```diff
@@ -20,7 +21,7 @@ import {CompilerInjectable} from './injectable';
 import {hasLifecycleHook} from './lifecycle_reflector';
 import {NgModuleResolver} from './ng_module_resolver';
 import {PipeResolver} from './pipe_resolver';
-import {ERROR_COMPONENT_TYPE, LIFECYCLE_HOOKS_VALUES, ReflectorReader, reflector} from './private_import_core';
+import {ERROR_COMPONENT_TYPE, LIFECYCLE_HOOKS_VALUES, ReflectorReader, reflector, viewEngine} from './private_import_core';
```
Comment (tbosch) [inline modules/@angular/compiler/src/metadata_resolver.ts]: revert

File: modules/@angular/core/src/application_module.ts [hunk 2]
```diff
@@ -37,20 +42,15 @@ export function _localeFactory(locale?: string): string {
...
+    Compiler, APP_ID_RANDOM_PROVIDER, ViewUtils, AnimationQueue,
     {provide: IterableDiffers, useFactory: _iterableDiffersFactory},
-    {provide: KeyValueDiffers, useFactory: _keyValueDiffersFactory},
-    {
+    {provide: KeyValueDiffers, useFactory: _keyValueDiffersFactory}, {
       provide: LOCALE_ID,
       useFactory: _localeFactory,
       deps: [[new Inject(LOCALE_ID), new Optional(), new SkipSelf()]]
     },
+    {provide: APP_INITIALIZER, useValue: _initViewEngine, multi: true}
```
Comment (tbosch) [inline modules/@angular/core/src/application_module.ts]: add `,` at end

File: modules/@angular/compiler/src/provider_analyzer.ts [hunk 3]
```diff
@@ -40,36 +45,39 @@ export class ProviderViewContext {
...
   private _allProviders: Map<any, ProviderAst>;
   private _attrs: {[key: string]: string};
   private _hasViewContainer: boolean = false;
+  private _queriedTokens = new Map<any, QueryMatch[]>();
 
   constructor(
       public viewContext: ProviderViewContext, private _parent: ProviderElementContext,
       private _isViewRoot: boolean, private _directiveAsts: DirectiveAst[], attrs: AttrAst[],
-      refs: ReferenceAst[], private _sourceSpan: ParseSourceSpan) {
+      private _refs: ReferenceAst[], private _sourceSpan: ParseSourceSpan) {
```
Comment (tbosch) [inline modules/@angular/compiler/src/provider_analyzer.ts]: make a var again

File: modules/@angular/compiler/src/template_parser/template_ast.ts [hunk 4]
```diff
@@ -241,6 +241,17 @@ export enum PropertyBindingType {
   Animation
 }
 
+export interface QueryId {
```
Comment (tbosch) [inline modules/@angular/compiler/src/template_parser/template_ast.ts]: Add comment: Only needs to be unique compared to all queries of children of an element.

File: modules/@angular/core/src/view/provider.ts [hunk 5]
```diff
@@ -70,7 +70,7 @@ export function _providerDef(
       outputDefs.push({propName, eventName: outputs[propName]});
     }
   }
-  const depDefs: DepDef[] = deps.map(value => {
+  let depDefs: DepDef[] = deps.map(value => {
```
Comment (tbosch) [inline modules/@angular/core/src/view/provider.ts]: revert

File: modules/@angular/core/src/application_module.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/core/src/application_module.ts]: Done

**Review by vicb — APPROVED** [review]

File: modules/@angular/compiler/src/provider_analyzer.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/compiler/src/provider_analyzer.ts]: Done

File: modules/@angular/compiler/src/template_parser/template_ast.ts (hunk 4 above)
Comment (tbosch) [inline modules/@angular/compiler/src/template_parser/template_ast.ts]: Done

File: modules/@angular/core/src/view/provider.ts (hunk 5 above)
Comment (tbosch) [inline modules/@angular/core/src/view/provider.ts]: Done

**Review by chuckjaz — APPROVED** [review]

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**